### PR TITLE
Add pension income and assets claims intake job

### PR DESCRIPTION
**Pull request by danlim715**
## Summary

- *This work is behind a feature toggle (flipper): YES*
- To support online submissions from the new Income and Assets form (21P-0969), we have added an intake job to process submissions and upload them to Lighthouse.
- Pension benefits team to own this work.

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/82558

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## What areas of the site does it impact?

Income and Assets

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

File: spec/controllers/v0/income_and_assets_claims_controller_spec.rb
Comment (wayne-weibel) [inline spec/controllers/v0/income_and_assets_claims_controller_spec.rb:1]: There was additional tests added in the pension module to get to 100% coverage that you could steal for here

**Review by wayne-weibel — CHANGES_REQUESTED** [review]

File: lib/income_and_assets/submissions/monitor.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ module IncomeAndAssets
   module Submissions
     class Monitor
       # statsd key for sidekiq
-      SUBMISSION_STATS_KEY = 'worker.income_and_assets_intake_job'
+      SUBMISSION_STATS_KEY = 'worker.lighthouse.income_and_assets_intake_job'
```
Comment (wayne-weibel) [inline lib/income_and_assets/submissions/monitor.rb:10]: does datadog reporting need to be updated?

**Review by wayne-weibel — CHANGES_REQUESTED** [review]
Some questions and documentation updates needed.  There are some failing tests as well
```
git diff origin/master...pension/add-income-and-assets-claim-intake --name-only -- '*.rb'
running yardoc ...
[warn]: @param tag has unknown parameter name: user_account_uuid 
    in file `app/sidekiq/lighthouse/income_and_assets_intake_job.rb' near line 36
[warn]: @param tag has unknown parameter name: user_uuid 
    in file `lib/income_and_assets/submissions/monitor.rb' near line 20
[warn]: @param tag has unknown parameter name: user_uuid 
    in file `lib/income_and_assets/submissions/monitor.rb' near line 40
[warn]: @param tag has unknown parameter name: upload 
    in file `lib/income_and_assets/submissions/monitor.rb' near line 40
[warn]: @param tag has unknown parameter name: user_uuid 
    in file `lib/income_and_assets/submissions/monitor.rb' near line 60
[warn]: @param tag has unknown parameter name: user_uuid 
    in file `lib/income_and_assets/submissions/monitor.rb' near line 79
[warn]: @param tag has unknown parameter name: user_uuid 
    in file `lib/income_and_assets/submissions/monitor.rb' near line 116
Files:           4
Modules:         4 (    3 undocumented)
Classes:         5 (    5 undocumented)
Constants:       3 (    2 undocumented)
Attributes:      0 (    0 undocumented)
Methods:        23 (    0 undocumented)
 71.43% documented
```

File: app/models/saved_claim/income_and_assets.rb [hunk 2]
```diff
@@ -5,11 +5,29 @@
...
+  ##
+  # send this Income and Assets Evidence claim to the Lighthouse Benefit Intake API
+  #
+  # @see https://developer.va.gov/explore/api/benefits-intake/docs
+  # @see Lighthouse::PensionBenefitIntakeJob
+  #
+  # @param current_user [User] the current user submitting the form
+  #
+  def upload_to_lighthouse(current_user = nil)
+    return unless Flipper.enabled?(:pension_income_and_assets_clarification, current_user)
```
[the remark this answers is not in the document]
Comment (danlim715) [reply app/models/saved_claim/income_and_assets.rb:29]: I don't believe so. But if we do need support for attachments, it shouldn't be too difficult to port over from pensions logic.

File: lib/income_and_assets/submissions/monitor.rb (hunk 1 above)
Comment (danlim715) [reply lib/income_and_assets/submissions/monitor.rb:10]: I don't think so. There shouldn't be any monitors or reports pulling I&A metrics just yet. Also, nothing in production is invoking these monitor functions just yet.

**Review by wayne-weibel — APPROVED** [review]
G2G

some documentation missing, but that can be addressed later, since this is still not live

**Review by ericboehs — DISMISSED** [review]

**Review by wayne-weibel — APPROVED** [review]

**Review by ericboehs — APPROVED** [review]

**Pull request merged**